nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIServiceImpl.java

```diff
@@ -213,4 +242,81 @@ public boolean verifyProofKey(String proofKeyHeader, String oldProofKeyHeader, S
         }
         return res;
     }
+
+    @Override
+    public boolean refreshDiscovery() {
+        byte[] discoveryBytes = fetchDiscovery();
+        if (ArrayUtils.isEmpty(discoveryBytes)) {
+            return false;
+        }
+
+        log.debug("Successfully fetched WOPI dicovery");
+        storeDiscovery(discoveryBytes);
+        loadDiscovery(discoveryBytes);
+        if (Framework.isBooleanPropertyTrue(CLUSTERING_ENABLED_PROP)) {
+            invalidator.sendMessage(new WOPIDiscoveryInvalidation(discoveryBytes));
```

Comment:
At this point `CLUSTERING_ENABLED_PROP` is an implementation detail, you should check `invalidator != null` instead.